web/slashes/scrapers.py

```diff
@@ -0,0 +1,218 @@
+import os
+import re
+import bs4
+import html
+from typing import List
+from flask import current_app, request
+
+from . import commands
+from ..executor import executor
+from ..utils import ForgetfulDict
+from .. import http, slack, formatting, utils
+
+
+class WebScrapingCommandHandler(commands.CommandHandler):
+    def __init__(self, *, relink_timeout):
+        super().__init__()
+
+        self.recently_linked_cache = ForgetfulDict(memory_time=relink_timeout)
+
+    def handle(self):
+        channel = request.form.get("channel_id")
+
+        args = []
+        skipped_args = []
+        requested_args = html.unescape(request.form.get("text")).split(" ")
+
+        for arg in requested_args:
+            if (arg.upper(), channel) not in self.recently_linked_cache:
+                self.recently_linked_cache[(arg.upper(), channel)] = True
+                args.append(arg)
+            else:
+                skipped_args.append(arg)
+
+        if len(args) == 0:
+            return (
+                f"Looked for {self.word}{formatting.plural(skipped_args)}"
+                + f" {', '.join(formatting.bold(k) for k in skipped_args)}"
+                + f" recently, skippping",
+                200,
+            )
+
+        user = request.form.get("user_id")
+        executor.submit(self.reply, channel, user, args)
+
+        message = f"Looking for {self.word}{formatting.plural(skipped_args)}"
+        message += f" {', '.join(formatting.bold(k) for k in args)}"
+        if len(skipped_args) != 0:
+            message += f", skipping recently-viewed {self.word}"
+            message += f"{formatting.plural(skipped_args)}"
+            message += f" {', '.join(formatting.bold(k) for k in skipped_args)}"
+        return (message, 200)
+
+    def reply(self, channel, user, args: List[str]):
+        response = http.cached_get_url(self.url)
+        soup = bs4.BeautifulSoup(response.text, "html.parser")
```

Comment:
Let's pull this out as a method so we can add caching later if desired.

Reply:
01b4499